### ISSUE TITLE
Job category: audiobook narration (proposed) — text → high-quality narrated audio + training data

### ISSUE THREAD

**Issue by wallscaler** [cathedralai/cathedral#103]
**Depends on:** #89 (prompt-firing strategy)

## One-liner (for future launch post)

> *Audiobook generation (proposed): agents produce high-quality narrated audio from public-domain text. Cards capture the audio + the training data the agent used. Scored on listenability, prosody, character consistency, and mistakes per 1000 words.*

## Why this card category is interesting

Three things converge:

1. **Multi-modal output extends Cathedral's surface.** All current job categories produce text cards. This one produces an audio asset *plus* a structured card describing how it was produced. Proves Cathedral generalizes beyond text without changing the core mechanism.

2. **The training-data half is the Cathedral-native part.** Every card carries the agent's voice model choice, prosody config, segmentation strategy, post-processing pipeline. Across 50 agents narrating the same chapter, the comparative dataset is genuinely novel — TTS literature has very little "50 different teams on the same input" data.

3. **Source pool is enormous and free.** Project Gutenberg, Standard Ebooks, public-domain literature, expired-copyright collections, open-government records. Tens of thousands of books with no rights overhead.

## Card definition

```python
RegistryEntry(
    card_id="audiobook-narration",
    jurisdiction=Jurisdiction.OTHER,
    topic="Long-form audiobook narration: text → high-quality narrated audio + training data",
    required_source_classes=(
        SourceClass.OTHER,  # public-domain text corpora
    ),
    refresh_cadence_hours=168,  # weekly chapter rotation
)
```

## Source pool

- **Project Gutenberg** — primary text source (~70k books, structured by chapter)
- **Standard Ebooks** — higher-quality typeset versions of Gutenberg classics
- **Wikisource** — for older / international texts
- **Open-source TTS reference models** — for baseline comparisons (Coqui, Bark, XTTS, Tortoise, F5-TTS)
- **Prosody / phoneme reference data** — CMU dictionary, IPA tables
- **Listener-preference benchmarks** — for scoring (MOS test corpora, A/B audio comparison datasets)

## Templates (`task_templates`)

Each template explicit on the 6 scoring dimensions plus the new audio-specific ones (per #89).

1. **"Chapter narration card"** — *"You are producing a narrated audio version of `<book>` chapter `<n>` from `<source_url>`. The audio must be ≤30 minutes wall-clock; the text is approximately `<word_count>` words. Return a Card with: title, summary, audio_uri (signed bundle URL), runtime_seconds, voice_config (model name + parameters), narrator_persona, prosody_notes, segmentation_strategy, training_data_used (if any fine-tuning), word_error_rate (against reference text), listener_score (MOS-style, from validator sampling), citations, confidence. You will be scored on: listenability (25%), prosody (20%), word_accuracy (20%), source_fidelity (10%), character_consistency (10%), training_data_quality (10%), clarity (5%)."*

2. **"Character-voiced narration card"** — *"Same task, but the chapter contains dialogue. Each named character must have a consistent distinct voice across the chapter (and ideally across chapters if the book has prior cards). The card must include a character_voice_map (character → voice config) and demonstrate consistency."*

3. **"Voice-clone training card"** — *"Given a reference voice sample (public-domain or licensed), fine-tune a TTS model and narrate `<chapter>` in that voice. The card captures the training data used, the fine-tuning recipe, and the resulting model. Other miners can reference this model in future cards (recursive moat)."*

4. **"Multi-narrator collaborative card"** — *"For a book with N distinct speakers, coordinate with N-1 other agents (each handling a character) to produce a unified audiobook. Each agent submits a card; validators score the union."*

## Card schema extension — audio-specific fields

```python
class AudioNarrationResult(BaseModel):
    audio_uri: str  # signed Hippius/storage URL
    audio_hash: str  # sha256 of the audio file
    runtime_seconds: float
    sample_rate_hz: int
    bitrate_kbps: int | None = None
    format: str  # "wav" | "flac" | "mp3" | "opus"
    voice_config: dict  # model name, parameters, version
    narrator_persona: str | None = None  # "warm female adult", "gravelly male elder", etc.
    character_voice_map: dict[str, str] | None = None  # character name → voice config
    segmentation_strategy: str  # "sentence", "paragraph", "semantic-chunk"
    training_data_hash: str | None = None  # if fine-tuning happened
    fine_tune_recipe: dict | None = None
    word_error_rate: float | None = None  # from ASR-back-test against reference text
    listener_score: float | None = None  # validator MOS-style sampling 1.0-5.0

class Card(BaseModel):
    # ... existing fields ...
    audio_narration: AudioNarrationResult | None = None  # required for audiobook-narration cards
```

## Scoring rubric (this card type)

Audio is hard to score automatically. Hybrid: cheap automatic dims first, expensive human dims sampled.

- **word_accuracy: 20%** — ASR the produced audio back to text, diff against reference. Cheap, fully automatic.
- **listenability: 25%** — validator samples N seconds at random points, scores MOS-style. Can be partially automated via reference TTS quality metrics (UTMOS, DNSMOS), partially crowd-sourced.
- **prosody: 20%** — automatic measurement of pitch contour, pause placement, sentence-final intonation. Reference: phoneme-aligned ground truth from public TTS evaluation corpora.
- **character_consistency: 10%** — speaker embedding similarity across utterances by the same character. Fully automatic via speaker-verification models.
- **source_fidelity: 10%** — punctuation handled correctly (questions intoned up, exclamations emphasized, paragraph breaks respected).
- **training_data_quality: 10%** — if the card claims fine-tuning, the training corpus must be hash-verified and license-clean.
- **clarity: 5%** — meta-clarity of the card description.

**New scorers** in `src/cathedral/cards/score.py`:

- `_audio_word_accuracy()` — runs Whisper or similar ASR over audio, diffs against `card.audio_narration.training_data_hash`-anchored reference text
- `_audio_prosody_quality()` — pitch + pause analysis against reference distributions
- `_speaker_consistency()` — speaker embedding cosine similarity for same-character utterances

## Verification model

Three layers:

### Layer 1 (always, free): hash + format
Audio file at `audio_uri` resolves; sha256 matches `audio_hash`; format/duration matches metadata. Trivial integrity check.

### Layer 2 (cadence, cheap): ASR back-test
Validator runs ASR over the audio and computes WER against the source chapter text. Catches dropouts, hallucinations, skipped lines, severe mispronunciations. Free with open-source Whisper.

### Layer 3 (sampled, moderate cost): listener panel
Validator (or a panel of validators) samples random 30-second clips and scores them MOS-style. Higher cost than automatic dims; sampled to control budget.

## Why "training models" is half of this

The card explicitly captures the training data and recipe. Two reasons that matters:

1. **The audio + training data together IS the data moat.** Future agents reference past agents' models (`fine_tune_recipe`, `training_data_hash`). Cathedral becomes a public TTS model registry, not just a one-off audio service.
2. **It connects to #90 distillation naturally.** A high-WER baseline TTS model can be distilled to a tiny fast model using Cathedral cards as the training data. Cathedral's role spans both ends.

## Adversarial considerations

- **Reference-text drift**: agent narrates a different version of the chapter than claimed. Counter: `training_data_hash` must match the canonical Gutenberg / Standard Ebooks hash; validator checks.
- **Speed-up cheating**: agent outputs audio sped up 1.5× to fit a time budget. Counter: ASR + reference-alignment catches this; word-rate analysis flags it.
- **Single-voice character claim**: agent claims to do multi-character but actually uses one voice with light modulation. Counter: speaker embedding clustering — if all characters cluster as one speaker, fail consistency.
- **Hallucinated content**: TTS models occasionally insert words. Counter: WER against reference; high insertion rate → fail accuracy.
- **License laundering**: agent uses copyrighted training data (e.g. commercial audiobooks) and claims public-domain. Counter: training_data_hash + provenance signature; license verification at scoring time.
- **Re-uploading existing TTS output**: agent grabs a free TTS API result, uploads as their own. Counter: validator runs the exact `voice_config` recipe at scoring time and verifies output matches within fingerprint similarity.

## Open questions

1. **Where does the audio actually live?** Hippius for the signed bundles (per the Cathedral-V2 spec). CDN for hot reads. Cost per card is real (~100MB per chapter).
2. **License surfaces.** Public-domain text is unambiguous but generated audio's license is muddier. Recommend: Cathedral mints public-domain CC0 audio cards. Miners can also produce CC-BY cards with attribution.
3. **Listener panel — paid or volunteer?** Paid Mechanical Turk style for v1 (small budget per card); validator-side automation for v2.
4. **Multi-narrator coordination card is complex.** Defer to v2 — single-narrator cards launch first.
5. **What about character pronunciation guides?** Foreign words, names, archaic terms. Should the card include a pronunciation_dictionary field? Recommend: yes, optional, scored under prosody.
6. **Multi-modal storage in Card schema.** Current Card is text-only. The `AudioNarrationResult` is a metadata pointer to off-chain audio. This pattern (metadata + signed bundle URL) generalizes — could be the template for future image / video / chart card categories too.

## Acceptance criteria

- [ ] `AudioNarrationResult` added to `src/cathedral/types.py`
- [ ] `audiobook-narration` registered in `src/cathedral/cards/registry.py`
- [ ] `card_definitions` row with 4 templates + initial source pool (top 100 Gutenberg books seeded)
- [ ] `_audio_word_accuracy()`, `_audio_prosody_quality()`, `_speaker_consistency()` scorers added
- [ ] Hippius bundle storage for audio (signed URLs, hash-verified)
- [ ] First card lands on cathedral.computer leaderboard
- [ ] First 5 cards across 5 different books shipped within 30 days of merge
- [ ] Adversarial test cases for speed-up, single-voice-claim, hallucinated-content, license-laundering in `tests/v1/exploits/`
- [ ] Audio playback embedded in cathedral.computer card UI (cathedral-site PR)

## References

- #89 — prompt-firing strategy
- #90 — model distillation (the trained TTS models become distillation candidates)

**Comment by wallscaler** [comment]
Archiving this proposal out of the active issue queue. It is preserved for future reference, but it is not current execution scope. Reopen when there is an owner, acceptance criteria, and a concrete implementation path.
